proposals/output_files.md

```diff
@@ -0,0 +1,52 @@
+# Proposal: Image ID, Digest, and Tar File Location Configuration
+
+Relevant issues: [#1561](https://github.com/GoogleContainerTools/jib/issues/1561), [#1921](https://github.com/GoogleContainerTools/jib/pull/1921)
+
+## Motivation
+
+Currently Jib hardcodes the location of the files it generates during the build, such as the image ID, digest,
+and tar file (for tar builds). For some users, it would be useful to allow configuring the location of these
+output files in a way that best fits their workflows.
+
+## Current Configuration
+
+Jib currently doesn't allow configuring these locations, and instead it uses hardcoded defaults:
+
+- Image tar -> `${buildDir}/jib-image.tar`
+- Image ID -> `${buildDir}/jib-image.id`
+- Image digest -> `${buildDir}/jib-image.digest`
+
+## Proposed Configuration
+
+The proposal is to allow users to configure their build with the following rules:
+1. Extensions to the filename (like id, digest, tar) will not be automatically appended.
+1. Existing files at the specified locations will be overwritten.
+1. Running `clean` will not delete output files created outside of the project's build directory.
+1. The configuration will accept both absolute and relative paths. Relative paths are resolved in the build tool's default manner.
+
+#### Maven (`pom.xml`)
+```xml
+<configuration>
+  <outputPaths>
+    <tar>/absolute/location.tar</tar>
+    <digest>relative/path/digest</digest>
+    <imageId>${project.build.directory}/id</imageId>
+  </outputPaths>
+</configuration>
+```
+
+#### Gradle (`build.gradle`)
+```groovy
+jib {
+  outputPaths {
+    tar = file("/absolute/location.tar")
```

Comment:
I think we should allow the user to use strings here. Gradle lets you accept `Object` and then resolve that yourself using `project.file()`. Like how we do in `ExtraDirectoryParameters.setPaths(Object paths)`

Reply:
I thought gradle auto-converts strings to paths by default? I thought the reason we used `Object` with `extraDirectories` was so we could accept both a single path or a list of paths.

Reply:
Yeah, I think we're saying the same thing. This example of gradle config uses `file()` everywhere though. Think one example with just `tar = "/absolute/location.tar"` might be helpful.

Reply:
Oh, sure.